plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/MissingComponentPage.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.ui;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.StyleRange;
+import org.eclipse.swt.custom.StyledText;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import com.google.cloud.tools.eclipse.usagetracker.AnalyticsEvents;
+import com.google.cloud.tools.eclipse.usagetracker.AnalyticsPingManager;
+
+/**
+ * Wizard page that displays a message that something needed in the environment is
+ * missing with instructions on how to install it. This page disables the 'Finish' button.
+ */
+public class MissingComponentPage extends WizardPage {
+
+  private String wizardType;
+  private String message;
+
+  public MissingComponentPage(String pageName, String wizardType, String title, String description,
```

Comment:
Could you add a comment here to indicate that wizardType is expected to be one of our wizard types on `AnalyticsEvents`. 

Also worth describing that message bolds text between single quotes.  Which doesn't seem a great choice, since it will catch any contractions and possessives too?  Perhaps we should change those to be markdown-like backquotes?